packages/sdk-cli/src/models/localRelay/util.test.ts

```diff
@@ -0,0 +1,72 @@
+import { promises as fsPromises } from 'fs';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { RELAY_PID_FILE_NAME } from './const';
+import { isInt, readJsonFile } from './util';
+
+// Have to do this, because 'fsPromises' __must__ be mocked (which sets all fs methods to jest.fn())
+jest.mock('fs', () => {
+  const actualFs = jest.requireActual('fs');
+  const actualFsPromises = actualFs.promises as typeof fsPromises;
+
+  return {
+    ...actualFs,
+    promises: {
+      ...actualFsPromises,
+      readFile: jest.fn().mockImplementation(actualFsPromises.readFile),
+    },
+  };
+});
+
+describe('isInt', () => {
+  describe('false', () => {
+    it.each([undefined, null, Infinity, -Infinity])('%s', (x) =>
+      expect(isInt(x)).toBe(false),
+    );
+
+    it('number string', () => expect(isInt('5')).toBe(false));
+  });
+
+  describe('true', () => {
+    it('number', () => expect(isInt(5)).toBe(true));
+  });
+});
+
+describe('readJsonFile – actual file system', () => {
+  it('reads file and returns JSON', async () => {
+    const json = { data: 'random' };
+    const contents = JSON.stringify(json, null, 2);
+
+    const path = join(tmpdir(), `readJsonFile-test-${RELAY_PID_FILE_NAME}`);
+    await fsPromises.writeFile(path, contents);
+
+    await expect(readJsonFile(path)).resolves.toEqual(json);
+
+    await fsPromises.unlink(path);
+  });
+
+  it("throws on file read error – file doesn't exist", async () => {
+    const path = join(tmpdir(), `readJsonFile-test-${RELAY_PID_FILE_NAME}`);
+    // Delete the file just in case
+    try {
+      await fsPromises.unlink(path);
+    } catch (error) {
+      if ((error as NodeJS.ErrnoException).code !== 'ENOENT') throw error;
+    }
+
+    await expect(fsPromises.open(path, 'r')).rejects.toMatchObject({
+      code: 'ENOENT',
+    });
+    await expect(readJsonFile(path)).rejects.toThrowError();
+  });
+
+  it('throws on file JSON parse error', async () => {
+    const contents = 'not json at all';
+    const path = join(tmpdir(), `readJsonFile-test-${RELAY_PID_FILE_NAME}`);
+    await fsPromises.writeFile(path, contents);
+
+    await expect(readJsonFile(path)).rejects.toThrowError();
+
+    await fsPromises.unlink(path);
```

Comment:
If the tests fails, this will never run and the file will be left on the filesystem. Things like this are why mock FS is a better idea as we discussed, it gives you a clean slate.

Reply:
Hm, I thought we agreed that it's okay to use real-fs in these tests. How guaranteed is it that `afterEach/All` will run, even if the test fails or unexpectedly throws an error? I can either move the cleanup to `afterEach` or just replace the real-fs tests with mock-fs

Reply:
I can't remember exactly what was said because it was a Meet call iirc, but I think the gist was either is okay, but both doesn't make sense, though mock FS gives a better clean slate for tests.

Reply:
`after{Each,All}` will be called even if the case fails as far as I know, it's code within the test body that may not depending where it fails.

Reply:
I would recommend mock FS.